torch/_functorch/aot_autograd.py

```diff
@@ -915,6 +919,8 @@ def fn_input_mutations_to_outputs(
     keep_data_input_mutations: bool,
 ) -> Any:
     def inner_fn(*args):
+        if config.functionalize_rng_ops:
+            PhiloxStateTracker.mark_beginning_of_forward()
```

Comment:
I'm not sure where the best place to put this is. Inside of `fn_input_mutations_to_outputs()` feels a little bad, because the idea was to factor out this logic to do only one thing (convert input mutations to outputs) and nothing else. But I see that for the joint case you need to insert markers for both the forward and backward (in `create_joint()`), so it looks like you're trying to put the logic in the same area for the inference case.

One alternative would be to put `PhiloxStateTracker.mark_beginning_of_forward()` in some shared code path for both inference + autograd, and only have the `mark_beginning_of_backward`()` in `create_joint`. I'm not sure that's better though.

Reply:
I prefer the current way, mostly because of readability.

Reply:
Why do you need to mark beginning of forward? Just implicitly assume you are in forward until told otherwise (mark beginning of backward).

Reply:
Good point!